src/components/Dialog.js

```diff
@@ -0,0 +1,139 @@
+import classnames from 'classnames';
+import { useEffect, useLayoutEffect, useRef, useState } from 'preact/hooks';
+
+import { IconButton, LabeledButton } from '../';
+import { SvgIcon } from './SvgIcon';
+
+let idCounter = 0;
+
+/**
+ * Return an element ID beginning with `prefix` that is unique per component instance.
+ *
+ * This avoids different instances of a component re-using the same ID.
+ *
+ * @param {string} prefix
+ */
+function useUniqueId(prefix) {
+  const [id] = useState(() => {
+    ++idCounter;
+    return `${prefix}-${idCounter}`;
+  });
+  return id;
+}
+
+/**
+ * @typedef {import("preact").ComponentChildren} Children
+ *
+ * @typedef DialogProps
+ * @prop {Children} [buttons] -
+ *   Additional `Button` elements to display at the bottom of the dialog.
+ *   A "Cancel" button is added automatically if the `onCancel` prop is set.
+ * @prop {string} [cancelLabel] - Label for the cancel button
+ * @prop {import("preact").ComponentChildren} children
+ * @prop {string} [contentClass] - CSS class to apply to the dialog's content
+ * @prop {string} [icon] - Name of optional icon to render in header
+ * @prop {import("preact/hooks").Ref<HTMLElement>} [initialFocus] -
```

Comment:
* @prop {import('preact').Ref<HTMLElement>} [initialFocus]

Reply:
This gives me `Generic type 'Ref' requires 1 type argument(s).ts(2314)`

Reply:
I meant `import('preact').Ref<HTMLInputElement>` but it is actually a different type. So forget about this suggestion.

The only thing maybe, would be to change the double quote for single quotes.